docs/aws/audit/redshiftmonitoring/rules/allow_cluster_version_upgrade.mdx

```diff
@@ -24,6 +24,217 @@ AWSWAF
+   ```
+   This command will return a list of all Redshift cluster identifiers.
+
+3. Check if Redshift clusters allow version upgrade: For each cluster identifier returned from the previous step, run the following command to check if the cluster allows version upgrades:
+
+   ```
+   aws redshift describe-clusters --cluster-identifier <ClusterIdentifier> --query 'Clusters[*].[AllowVersionUpgrade]' --output text
+   ```
+   Replace `<ClusterIdentifier>` with the actual cluster identifier. This command will return a boolean value. If it returns `true`, it means the cluster allows version upgrades. If it returns `false`, it means the cluster does not allow version upgrades.
+
+4. Repeat step 3 for all clusters: You need to repeat the third step for all the cluster identifiers returned from the second step. This will help you check if all Redshift clusters in your account allow version upgrades.
+</Accordion>
+
+<Accordion title='Using Python'>
+To check if Redshift Clusters allow version upgrade in AWS using Python scripts, you can use the Boto3 library, which allows you to write software that makes use of services like Amazon S3, Amazon EC2, etc. Here are the steps:
+
+1. **Import the Boto3 library in Python:**
+   Boto3 is the Amazon Web Services (AWS) Software Development Kit (SDK) for Python, which allows Python developers to write software that makes use of services like Amazon S3, Amazon EC2, etc. You can install it using pip:
+   ```python
+   pip install boto3
+   ```
+
+2. **Create an AWS session using your credentials:**
+   You need to configure your AWS credentials. You can do this in several ways, but the simplest is to use the AWS CLI. Here is a sample script:
+   ```python
+   import boto3
+
+   session = boto3.Session(
+       aws_access_key_id='YOUR_ACCESS_KEY',
+       aws_secret_access_key='YOUR_SECRET_KEY',
+       region_name='us-west-2'
+   )
+   ```
+
+3. **Get the list of all Redshift clusters and check their 'AllowVersionUpgrade' attribute:**
+   Use the `describe_clusters` method to get a list of all Redshift clusters. Then, iterate over each cluster and check the 'AllowVersionUpgrade' attribute. If it's set to False, then the cluster doesn't allow version upgrades.
+   ```python
+   redshift = session.client('redshift')
+
+   clusters = redshift.describe_clusters()
+   for cluster in clusters['Clusters']:
+       if not cluster['AllowVersionUpgrade']:
+           print(f"Cluster {cluster['ClusterIdentifier']} does not allow version upgrades.")
+   ```
+
+4. **Handle exceptions:**
+   It's a good practice to handle exceptions in your scripts. You can do this by using try/except blocks. For example, you can catch the `botocore.exceptions.NoCredentialsError` exception if the script can't find your AWS credentials.
+   ```python
+   try:
+       # Your code here
+   except botocore.exceptions.NoCredentialsError:
+       print("No AWS credentials found.")
+   ```
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>